packages/devtools-kit/src/core/component/state/editor.ts

```diff
@@ -88,26 +88,23 @@ export class StateEditor {
   }
 }
 
-class RefStateEditor {
+export class RefStateEditor {
   set(ref: Ref<any>, value: any): void {
     if (isRef(ref)) {
       ref.value = value
     }
     else {
       // if is reactive, then it must be object
       // to prevent loss reactivity, we should assign key by key
-      const previousKeys = Object.keys(ref)
+      const previousKeysSet = new Set(Object.keys(ref))
       const currentKeys = Object.keys(value)
       // we should check the key diffs, if previous key is the longer
       // then remove the needless keys
-      // @TODO: performance optimization
-      if (previousKeys.length > currentKeys.length) {
```

Comment:
The origin logic can not pass this test case:
https://github.com/vuejs/devtools-next/pull/221/files#diff-5ff2b319492619782280fbf567ced9d18260b94c2c3aa8e8f9e865a0b45ab07bR19